Article suit parts: blobs

NeetBux-Hash opened the pull request:
A little tutorial on blobs. I think its clear enough if people got through the previous tutorials. No indexing or any other files were touched because I lack experience

NeetBux-Hash committed:
Rough Blobs intro

No markdown styling

NeetBux-Hash committed:
Blobs article done

NeetBux-Hash committed:
Refining blobs

Made a little tutorial for making blobs

NeetBux-Hash committed:
Article no advanced search

NeetBux-Hash committed:
Needs way more refinement like blobs

Compare to saved txt file. Lots of lost details and clarification

NeetBux-Hash committed:
Halfway nowhere

This article needs replacing

NeetBux-Hash committed:
Replacing this next

NeetBux-Hash committed:
Good but needs checking and Sidebar

atampy25 commented on docs/modding/suitmodding/addinglocals.md:
```diff
@@ -0,0 +1,186 @@
+            "3718104824": "The Blood Money suit but with extra style."
+        },
+        "french": {
+            "1047194709": "Costume d'argent du sang avec goutte à goutte",
```
This line means "Blood money suit with IV drip" I believe

atampy25 commented on docs/modding/suitmodding/addinglocals.md:
```diff
@@ -0,0 +1,186 @@
+
+---
+
+This version is more structured, gradually introducing the steps with clear explanations. Let me know if this feels clearer and easier to follow!
```
Did ChatGPT write that part or

NeetBux-Hash replied:
Yes. I wrote the article myself but the markdown syntaxing was done with gpt.

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
```diff
@@ -0,0 +1,64 @@
+
+## Step 1: Locate the Suit in RPKG
+
+1. Open **RPKG** 
```
I would recommend GlacierKit over RPKG wherever possible

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
```diff
@@ -0,0 +1,64 @@
+sidebar_position: 3
+description: Modifying an image in the menu
+---
+To continue on the adding parts page, you'll likely also want to modify the image in the menu to show the difference, change the suit's name and even add in your own suit's descritpion of it. I'll also mention a potential issue that arises when adding parts to your suit that dont come from a similiar chunk which will have to be addressed in the manifest. If you haven't read the documentation on m ...
```
`similiar` -> `similar`, `Suggest` -> `suggest`

atampy25 commented on docs/modding/suitmodding/addinglocals.md:
```diff
@@ -0,0 +1,186 @@
+To improve clarity and structure in the article, I’ll organize the information in a more logical flow. The goal will be to avoid overwhelming the reader with too much technical detail upfront and to guide them step-by-step, with each section focusing on one aspect of the process. Here's a revised version:
```
That's great ChattyG but irrelevant to this article

NeetBux-Hash replied:
This shouldn't be part of the article

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
```diff
@@ -0,0 +1,64 @@
+
+---
+
+# Part 1: Blobs
```
Part 1?

NeetBux-Hash replied:
Yes for the article after this one. My introduction up top was meant to prepare you for multiple articles

atampy25 commented on docs/modding/suitmodding/addinglocals.md:
```diff
@@ -0,0 +1,186 @@
+
+---
+
+# Part 2: Localisation Overrides
```
Part 2?

NeetBux-Hash replied:
I'll just remove the parts maybe. The point was to smooth the reader from the adding parts to the next articles

atampy25 commented on docs/modding/suitmodding/addinglocals.md:
```diff
@@ -0,0 +1,186 @@
+
+In this section, we'll focus on changing the **suit name and description** in the game by modifying the localisation files. You will need **GlacierKit** and a mod folder with a manifest file, but no content files are required. Everything will be handled through the manifest.
+
+If you’re editing the manifest using **GlacierKit**, it will auto-suggest code to keep things clean and help reduce errors.
```
VS Code does the same thing if you have the manifest schema line

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
````diff
@@ -0,0 +1,64 @@
+   ```
+4. Remember the **file name** and **file path** ("images/unlockables_override/" in this case).
+
+> Keep RPKG open, you’ll need it in the next steps.
````
There are proper admonition blocks you can use instead of markdown quotes, examples in basicretexture

NeetBux-Hash replied:
I'll take a look at them after I know what admonition is

atampy25 commented on docs/modding/suitmodding/addinglocals.md:
````diff
@@ -0,0 +1,186 @@
+     ```
+   - If you’re using GlacierKit, it will handle much of the code suggestions and format checks.
+
+> **Note:** It might seem overkill to install Visual Studio Code for this tutorial, but you’ll appreciate the cleaner interface and error-checking features as you continue modding.
````
Same as before with admonition blocks

atampy25 commented on docs/modding/suitmodding/addinglocals.md:
```diff
@@ -0,0 +1,186 @@
+
+---
+
+## Step 3: Convert the String to a Decimal Number
```
Hash is a more accurate term than number

NeetBux-Hash replied:
Ill stick to using "strings" for now just to clearly differentiate the locr hash and the hash inside the locr that needs to be converted by calling it a "string". Itll make sense if you read the full article. I will change it later if needed

NeetBux-Hash commented:
Aologies for all the mistakes. I'm already working on an improved version and another article about localisation overrides that should wrap things more up. This pull request might have been too premature but I felt it was good enough to present it as a draft. Appreciate the feedback

Notexe commented:
There’s no need to apologise for anything, you are good.

NeetBux-Hash committed:
Finished Blobs.

Still needs a proper index of some sorts. it just shows the filename on top currently. The introduction might need some kind of cleaning or header too

NeetBux-Hash committed:
minor typo

NeetBux-Hash committed:
Blobs done. Maybe too much adminations

NeetBux-Hash committed:
More Blobs article refining

NeetBux-Hash committed:
More refining of blobs article

NeetBux-Hash committed:
Ignore locals

NeetBux-Hash committed:
blobs refining

NeetBux-Hash committed:
Finishing blobs and adding raw local article

NeetBux-Hash committed:
blobs refining

Still visually too much admonitions

NeetBux-Hash committed:
Blobs refining

NeetBux-Hash committed:
Blobs considered done

NeetBux-Hash committed:
adding raw locals back in

NeetBux-Hash commented:
Blobs is def ready for review

NeetBux-Hash committed:
deleting old raw non md

NeetBux-Hash committed:
More refining when looking on mobile

NeetBux-Hash committed:
blobs refining

NeetBux-Hash committed:
blobs refining

NeetBux-Hash committed:
refing blobs

NeetBux-Hash committed:
Startting on locals

Work on order of explos in first part

NeetBux-Hash committed:
return locals ai

Notexe commented on docs/modding/suitmodding/addingblobs.md:
```diff
@@ -0,0 +1,234 @@
+   ### 2. Locate the assembly path
+There will be a line of code starting with `"Image":"....`
+
+   :::example
```
```
Warning:  Docusaurus found 1 unused Markdown directives in file "docs/modding/suitmodding/addingblobs.md"
- :::example (106:4)
```

I think maybe having it tabbed is causing this error. Not really too sure though.

NeetBux-Hash replied:
I will just remove it since it's literally part of the tutorial. I thought example admonition was real

NeetBux-Hash committed:
Update addingblobs.md removing example admonition

NeetBux-Hash committed:
Blobs refining

Getting rid of quotation marks and setting a single form of highlightiing for same keywords. SHortening suggestion sentences and imprving of clarity of sentences

NeetBux-Hash committed:
locals update

NeetBux-Hash committed:
needs way more restructuring of info

NeetBux-Hash committed:
AI airticle is clean but no advanced search

NeetBux-Hash committed:
MDX file attempt

My implementation of a tooltip.
Its a mess to read or even attempt trying to correct it. I think it was worth it.
Have to alter bold not getting underlined
And some padding on tooltips
More styling maybe too

Notexe commented:
I'm assuming that this PR is very far from being ready considering there is `addinglocalsRAWautoMD.md` and `addinglocalsRAWautoMD copy.mdx`? I'm going to convert it back into a draft for now.

NeetBux-Hash committed:
commit this

NeetBux-Hash commented:
The blobs article was as good as done. I shouldn't have worked on both articles on a single PR. Maybe I'll try adding some tooltips there as well

NeetBux-Hash committed:
Before css fix attempt

NeetBux-Hash committed:
Good tooltips text set

The tooltips before hovering are too distracting yet not indistinguishable enough without it

NeetBux-Hash committed:
Before adding css icon class

NeetBux-Hash committed:
Working tooltip icon

NeetBux-Hash committed:
Commiting before changing java

NeetBux-Hash committed:
refinig locals

NeetBux-Hash committed:
half baked attempt of a dynamical tooltip

NeetBux-Hash committed:
white border tooltip

NeetBux-Hash committed:
green border, grey bg tooltip

NeetBux-Hash committed:
subtle glow

NeetBux-Hash committed:
tooltip dynamic refining

Getting close but still acting random

NeetBux-Hash committed:
tooltip always bottom

Cant figure out the navbar from blocking it

NeetBux-Hash committed:
tooltip back to top

No reason blocking the content following up

NeetBux-Hash committed:
removed icon by accident in css

NeetBux-Hash committed:
before removing locals overview

NeetBux-Hash committed:
cleaning redundant tooltips, adding checkered glossary

NeetBux-Hash committed:
darker check padding

NeetBux-Hash committed:
Proper glossary

NeetBux-Hash committed:
Forgot a key example in part 2

NeetBux-Hash committed:
fixed simplified locr

NeetBux-Hash committed:
adding ai adv search. To be fixed

NeetBux-Hash committed:
adding extra advanced link at suits. Might remove hash

NeetBux-Hash committed:
correctliy imported tippy.js

NeetBux-Hash committed:
Updated tooltip

 with grey bg and working arrow and nicer border

NeetBux-Hash committed:
Updated tooltip in css and js to use custom style

NeetBux-Hash committed:
added underline back

NeetBux-Hash committed:
Updated tooltip style to rounded

NeetBux-Hash committed:
refining locals article

NeetBux-Hash committed:
Committing before restructuring part 2

NeetBux-Hash committed:
Much shorter and consice version RAW

Needs tooltip, english and flow adjustments of info

NeetBux-Hash committed:
shorter guide refining base

Might use this instead. If information overload can be elminiated
We use it

NeetBux-Hash commented:
In commit [tippy](https://github.com/glacier-modding/glaciermodding.org/pull/32/commits/551961b0444709a911f04e7d3c48d91dfe278571)
I added tippy via npm. Now this changed some core files with react. The site is still functional (using yarn start --host on mobile as well). Just a warning bc idk if the pull request if succeeded can break something. As far as i can see, it only added a single line of code to a yarn document

Notexe commented:
> In commit [tippy](https://github.com/glacier-modding/glaciermodding.org/pull/32/commits/551961b0444709a911f04e7d3c48d91dfe278571)
> 
> I added tippy via npm. Now this changed some core files with react. The site is still functional (using yarn start --host on mobile as well). Just a warning bc idk if the pull request if succeeded can break something. As far as i can see, it only added a single line of code to a yarn document

Would have been better adding that dependency using yarn instead. What is it needed for anyways?

NeetBux-Hash commented:
> > In commit [tippy](https://github.com/glacier-modding/glaciermodding.org/pull/32/commits/551961b0444709a911f04e7d3c48d91dfe278571)
> > I added tippy via npm. Now this changed some core files with react. The site is still functional (using yarn start --host on mobile as well). Just a warning bc idk if the pull request if succeeded can break something. As far as i can see, it only added a single line of code to a yarn document
> 
> Would have been better adding that dependency using yarn instead. What is it needed for anyways?

Idk if you can view my latest articles but its just a tooltip implementation to massively shorten my guides. Idk too if you yarn start my articles before you attempted and corrected mistakes before, but trying to see or even try correcting my articles from the md files is almost impossible. The tooltips hopes to fix both the length of the article and the code

NeetBux-Hash committed:
SHortened copy refining

NeetBux-Hash committed:
refinig mini locals

Notexe commented:
> PS im checking the commit now and it does seem to be just a single line added to a yarn or yml file but package locks changing is what got me worried

That's because you used npm to install the dependency. This project is using yarn so you would have had to run `yarn add tippy.js` instead.

NeetBux-Hash committed:
refining mini locals

NeetBux-Hash committed:
refining locals mini

NeetBux-Hash committed:
refining custom tooltip exclam icon

NeetBux-Hash committed:
More icon refining

NeetBux-Hash committed:
refining locals

NeetBux-Hash committed:
commiting locals

NeetBux-Hash committed:
refining locals

NeetBux-Hash committed:
refining locqls

NeetBux-Hash committed:
refining locals

glossary and admonitions needs refining as well in resiurces tab

NeetBux-Hash committed:
refinig locals

NeetBux-Hash committed:
refining local

NeetBux-Hash committed:
adding condensed draft of short locals

NeetBux-Hash committed:
backup before deleting package lock

NeetBux-Hash committed:
deleted package lock json

Tooltip seems to be working fine after reloading live page from yarn start. Nothing suspicious or broken going on

Notexe commented:
I'm confused why there is `addinglocals.mdx` and `addinglocals copy.mdx`

atampy25 commented on .yarnrc.yml:
```diff
@@ -1 +1,3 @@
+nodeLinker: node-modules
```
Should the node linker be changed? Currently the website seems to be using yarn PnP

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
```diff
@@ -0,0 +1,253 @@
+description: Modifying an image in the menu
+
+---
+# Adding "Blobs"
```
Quotes are not necessary, blobs are what they are called

NeetBux-Hash replied:
I did that to introduce an unfamiliar term. I'll change it

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
```diff
@@ -0,0 +1,253 @@
+4. Open it by clicking on it (obviously) and then click on *Open in editor* in the new menu that popped up on the right. This is the **repository** but in **Glacierkit**
+
+:::tip
+You now know the exact name of the **repository** file. Next time you want to open the **repository** in **GlacierKit** just type ***pro.repo***.
```
This tip is just restating step 3 but saying not to do part of step 3, you can just make step 3 "search pro.repo" instead of "copy this hash and then paste and hit enter but actually don't do that use pro.repo instead"

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
````diff
@@ -0,0 +1,253 @@
+   ```
+   and its **assembly path**
+   ```json
+   images/unlockables_override/47_outfits_bloodmoney_gloves.jpg
````
This is not an assembly path, it's a blob path. The assembly path for this blob is much longer because blobs are inside the `cloudstorage` folder

atampy25 replied:
Also this should not be a JSON code block

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
```diff
@@ -0,0 +1,253 @@
+
+:::info GlacierKit vs. RPKG
+While **RPKG** is helpful for viewing images in the **repository**, it’s best to work primarily in **GlacierKit**, as it will be your main modding tool. The **repository code lines** in both tools are identical for each file, so practicing in **GlacierKit** is encouraged for efficient modding.
+:::
```
I do think it'll be a little confusing to show two separate tools for doing something and then say "it's best to not do the first one we told you to do"

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
```diff
@@ -0,0 +1,253 @@
+## STEP 2: **FIND THE IMAGE**
+In this step we will have to look for the correct image code and find its **assembly path**.
+:::note
+You already did this step in the **GlacierKit** method but follow along
```
What? Why is this repeated then

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
```diff
@@ -0,0 +1,253 @@
+
+It consists of a
+- **file path** `images/unlockables_override/`
+- **file name** `47_outfits_bloodmoney_gloves.jpg`
```
The whole thing, including the file name, is the file path

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
```diff
@@ -0,0 +1,253 @@
+## STEP 3: **SET UP A BLOBSFOLDER**
+In this step, you’ll create a custom folder structure that ***recreates*** the default image's **assembly path**.
+
+### 1. Make a blobsfolder
```
```suggestion
### 1. Make a blobs folder
```

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
````diff
@@ -0,0 +1,253 @@
+   blobs/images/unlockables_override/
+   ```
+### 3. Add your custom image  
+   - Place your image in the last folder you created (in this example, `unlockables_override`). 
````
I can't see this fully previewed obviously but I feel like this formatting will look odd, having a big size-3 heading with a dot point; usually you'd just have body text instead of placing the step in a dot point (also the previous steps don't use dot points).

NeetBux-Hash replied:
Yes, not being able to preview the site is a problem. Lots of choices are made aren't reflected properly from reviewing the MD file. I was looking into showing and sharing a link to a build version

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
````diff
@@ -0,0 +1,253 @@
+   blobs/images/unlockables_override/47_outfits_bloodmoney_gloves.jpg
+   ```
+### 5. Confirm the assembly path
+This fully recreates the **assembly path** above of the image you're replacing.    
````
Blob path

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
````diff
@@ -0,0 +1,253 @@
+   ```
+:::danger Important
+- **Aspect Ratio**: Match the aspect ratio of your replacement image with the original to avoid distortion. 
+>*Example*: The Blood Money suit image has a 4:3 ratio (resolution 696x520), which you can check in **GlacierKit**.
````
This can just be "for example", you don't really need so much emphasis and a separate quote block

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
```diff
@@ -0,0 +1,253 @@
+## STEP 4: **UPDATE MANIFEST**
+
+### 1. Open your manifest
+Recommended to do so in **GlacierKit** (or in **VS code** with a ***"schema"***).
```
There is a huge amount of bold and italic.
Also quotes around schema are not necessary, that is what it's called (might be a good idea to add a tooltip explaining what a schema is? or a link to somewhere else it's explained?)

NeetBux-Hash replied:
Agree, Blobs article will need a rework with a tooltip. That's the copy of the locals article. A draft for the current locals artoce but condensing it and changing the flow of information

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
```diff
@@ -0,0 +1,253 @@
+### 1. Open your manifest
+Recommended to do so in **GlacierKit** (or in **VS code** with a ***"schema"***).
+
+### 2. Add your blobsfolder:
```
```suggestion
### 2. Add your blobs folder
```

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
```diff
@@ -0,0 +1,253 @@
+Recommended to do so in **GlacierKit** (or in **VS code** with a ***"schema"***).
+
+### 2. Add your blobsfolder:
+Add the following line that will make **SMF** do a ***blobs operation*** and have it refer to your ***`blobs`*** folder
```
This line is just words

```suggestion
Add the following line, so that SMF recognises your blobs folder and adds the image to the game:
```

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
````diff
@@ -0,0 +1,253 @@
+   ```json
+   "blobsFolders": ["blobs"]
+   ```
+>Here’s how your full manifest file might look with a `blobsFolders` line added:
````
Quote?

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
````diff
@@ -0,0 +1,253 @@
+}
+   ```
+:::info Schema
+Notice the `$schema` line at the top. This line helps **VS Code** recognize the structure and options available for the manifest, providing error-checking and code suggestions as you edit. **GlacierKit** includes these features by default, but installing **VS Code** is highly recommended for better flexibility and ease in editing complex files.
````
Oh schemas are explained here, then you don't need to link/quote it earlier

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
```diff
@@ -0,0 +1,253 @@
+
+---
+## More on Blobs
+These tips will help you manage multiple images and **blobsfolders**.
```
```suggestion
These tips will help you manage multiple images and blobs folders.
```

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
```diff
@@ -0,0 +1,253 @@
+- If you want to add more images, remember to recreate the **file paths** by making more folders. 
+And place each image in its relevant folder
+
+examples:
```
```suggestion
Some examples:
```

atampy25 commented on docs/modding/suitmodding/addingblobs.md:
```diff
@@ -0,0 +1,253 @@
+| `images/characters/male_suits/caruso_idk.png`            | `blobs/images/characters/male_suits/caruso_idk.png`       |
+| `icons/equipment/distraction_device_old.jpg`               | `blobs/icons/equipment/distraction_device_old.jpg`          |
+
+- For multiple images that have the same **file path** in their **assembly path**, simply place them within the same folders (e.g., images under `unlockables_override`). Don't forget to **rename them too**.
```
This line basically doesn't make sense (file path? assembly path?) unless you have the definitions earlier, which are not correct. Also rename them to what?

atampy25 commented on docs/modding/suitmodding/addinglocals copy 2.mdx:
```diff
@@ -0,0 +1,92 @@
+---
```
I'm gonna assume these copy files are just drafts written by ChatGPT and not intended to be on the website?

NeetBux-Hash replied:
The copy files are me making some changes to when new information gets introduced. Like in the blobs folder it's all over the place so as not to overwhelm the user with technical jargon and kind of reel them in the process while they're learning stuff

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+
+
+<details>
+<summary>**About Tooltips**</summary>
```
I feel like this whole section can be condensed to "Tooltips are included throughout the article. You can hover over them for a quick explanation."

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+
+>By the end, you'll have a fully customized suit name and description in the game.
+
+*When necessary, there will be a link to the ***advanced search section*** to change any other text as well*
```
What?

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+</details>
+
+You will need: 
+-  <Tooltip text={<strong>GlacierKit</strong>} tooltip="A modding tool used for editing and managing game files in Hitman: World of Assassination." />
```
This terminology should have been introduced long before the user gets to this tutorial, so I can't see a tooltip being super useful here

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+<summary>about **HASH**</summary>
+:::info HASH
+You can interpret <Tooltip text={<strong>HASH</strong>} tooltip="A unique identifier that acts as the address or coordinates for a file." /> lines as a file's specific address or coordinates. They are the file's *unique identifiers* and can bring us straight to the file.
+>You will constantly have to use these throughout modding.
```
This is an odd explanation, also HASH is not an acronym

atampy25 replied:
Some terminology like this is explained here: https://hitman-resources.netlify.app/documentation/terminology/hash

NeetBux-Hash replied:
I'll keep it brief and add a link

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+- copy  your relevant <Tooltip text={<strong>HASH</strong>} tooltip="A unique identifier that acts as the address or coordinates for a file." />  
+- Go to the <Tooltip text={<strong>game contents tab</strong>} tooltip="A tab in Glacierkit where you search game files" />
+- paste it on the search bar and hit enter.
+>You'll see a map tree or *file path* leading to a <Tooltip text={<><strong>LOCR File</strong><span className="icon-alert"></span></>} tooltip="A file that holds game text in different languages, linked by unique 'strings'." />.
```
A map tree?

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+
+## Pt. 2: **GET YOUR STRING**
+*Every line of code in a* <Tooltip text={<strong>LOCR File</strong>} tooltip="A file that holds game text in different languages, linked by unique 'strings'." /> *consists of a* 
+>   - <Tooltip text={<><strong>string</strong><span className="icon-alert"></span></>} tooltip= "A key that links to text in the LOCR File" /> 
```
Technically a localisation string is the actual text in English/French/whatever? The distinction is not super important but iirc the game calls the `UI_TEXT_XYZ` "localisation keys"

NeetBux-Hash replied:
I'll use that term then instead of strings. But a distinction had to be made between the two hash types of the file and the "strings" inside

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+---
+
+## Pt. 3: **CONVERT TO DECIMAL**
+*We need to convert the* <Tooltip text={<strong>string</strong>} tooltip= "A key that links to text in the LOCR File" /> *to a* <Tooltip text={<><strong>decimal number</strong><span className="icon-alert"></span></>} tooltip="A string converted into a numeral form" /> *to make it usable in the manifest file*.
```
These tooltips are very duplicated, surely there's some better way of doing this in code

NeetBux-Hash replied:
There probably is but idk how really. Haven't seen a tooltip on any of the articles on the site and I'm new to coding. Probably the overuse of tooltips and using bold in general is the problem

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+
+
+### 1. Go to Text Tools 
+- Go to the <Tooltip text={<strong>Text Tools</strong>} tooltip="A tab in GlacierKit that provides tools for converting text and hashes for game modifications." /> tab
```
Technically a panel not a tab since it's on the left but doesn't really matter

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+- Copy *only* the converted <Tooltip text={<strong>decimal number</strong>} tooltip="A string converted into a numeral form" />. 
+
+>*We now have everything needed for the next part:*
+ >- a <Tooltip text={<strong>decimal form</strong>} tooltip="A string converted into a numeral form" /> of our <Tooltip text={<strong>string</strong>} tooltip= "A key that links to text in the LOCR File" />
```
The "decimal form" is in fact also a hash (because all a hash is is a piece of text calculated from another piece of text), which is confusing I know but people should probably be told about to avoid further naming confusion.

Ideally we would be calling these "localisation hashes" and the ones used to refer to game files "RuntimeIDs" or "RuntimeResourceIDs"; unfortunately RPKG tool took the name hash and applied it to literally everything possible, including calling game resources "hash files"

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+- *Method 2*:
+    - manually typing it in, starting with `"`
+    - navigate the *suggestions menu* to find `localisationOverrides`.
+    
```
This is just basic text editing, and I feel like if they cannot distinguish between copy/paste and manual typing there will be other challenges before they get to this point

NeetBux-Hash replied:
Trying to find a sweet spot for noobs and intermediate users was the balance I tried to keep. Which is why there is also weird distinguishing terms like file path and file name.

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
````diff
@@ -0,0 +1,666 @@
+    > "localisationOverrides": {}
+    >  ```
+:::tip using "suggestions"
+*Now you know how to use the* <Tooltip text={<strong>suggestions feauture</strong>} tooltip="This feauture is only supported in Glacierkit or in VS code with a schema." />. It will auto-complete most of the tedious functions in the steps to follow like:
````
`feauture` -> `feature`

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+:::info Syntax
+If you're paying attention, you might notice that we're setting up the localisation override code in the manifest just like our <Tooltip text={<strong>LOCR File</strong>} tooltip="A file that holds game text in different languages, linked by unique 'strings'." /> 
+where we got the <Tooltip text={<strong>strings</strong>} tooltip= "A key that links to text in the LOCR File" /> from. 
+Both will be using the same <Tooltip text={<strong>'Syntax'</strong>} tooltip="A set of rules that define how code is written and organized." />.
```
Syntax is also just what it's called, doesn't need quotes

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+where we got the <Tooltip text={<strong>strings</strong>} tooltip= "A key that links to text in the LOCR File" /> from. 
+Both will be using the same <Tooltip text={<strong>'Syntax'</strong>} tooltip="A set of rules that define how code is written and organized." />.
+
+> This was visually represented as a *simplified LOCR file* at the [beginning of Part 2](#pt-2-get-your-string).
```
What?

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+                "yourconverteddecimal": "New Suit Name"
+            },
+            "french": {
+                "yourconverteddecimal": "Niveua Costume nome"
```
Whatever language this is, it's definitely not French

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+
+:::info language codes
+- The reason why `french` is used here and not `français` is because the manifest's <Tooltip text={<strong>schema</strong>} tooltip="A structure used to validate and provide auto-completion for your manifest, helping you write error-free code." /> is coded in English. 
+- As for why it doesn't use the same <Tooltip text={<strong>language codes</strong>} tooltip="The language identifiers in each language array of a LOCR file" /> as the actual <Tooltip text={<strong>LOCR Files</strong>} tooltip="A file that holds game text in different languages, linked by unique 'strings'." /> like `en` or `fr`, is because the manifest's <Tooltip text={<strong>schema</strong>} to ...
```
"As for why it doesn't use the same as the other file, that's because that's the way it is" is not exactly super helpful. I don't really have a better explanation but this whole block doesn't really add anything

NeetBux-Hash replied:
Well I just had to find a way to emphasize it. I can remove the reason for why it its the way it is after

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+<summary>Language Codes</summary>
+
+:::info language codes
+- The reason why `french` is used here and not `français` is because the manifest's <Tooltip text={<strong>schema</strong>} tooltip="A structure used to validate and provide auto-completion for your manifest, helping you write error-free code." /> is coded in English. 
```
I feel like they would have noticed the manifest is written in English a long term before this point

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
````diff
@@ -0,0 +1,666 @@
+        }
+    }
+```
+>*This recreates how the* <Tooltip text={<strong>LOCR File</strong>} tooltip="A file that holds game text in different languages, linked by unique 'strings'." /> *is internally set up by using the same* <Tooltip text={<strong>syntax</strong>} tooltip="A set of rules that define how code is written and organized." />.
````
While it would be nice if it did, the manifest does not use the same syntax as the LOCR file, that's why you've had to explain how to use decimal hashes and that it's `english` not `en`

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+::: 
+
+---
+## More on LocalisationOverrides
```
```suggestion
## More on localisation overrides
```

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+
+*If you need to modify more* <Tooltip text={<strong>LOCR Files</strong>} tooltip="A file that holds game text in different languages, linked by unique 'strings'." />: 
+
+1. **Insert a Comma**: After your previous <Tooltip text={<strong>LOCR Array</strong>} tooltip="an array in the manifest resembling the game's LOCR file syntax" />  in the <Tooltip text={<strong>manifest</strong>} tooltip="A file in your mod folder that holds important information and can execute code" />.
```
LOCR array? What?

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+1. Clicking on the <Tooltip text={<strong>Advanced Search Tab</strong>} tooltip="A tab in GlacierKit for performing extended searches. Located on the left." />
+
+2. **Set Your Search Parameters**:
+   - Check only the `chunk0` box.
```
This is not generally good advice, while suit localisation is in chunk0 you don't want people to end up thinking of chunk0 as a magical box they just need to tick, because then they'll try to search for things in Mendoza and fail

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+     - "Blood Money"
+     - "blood money"
+     - "BLOOD MONEY"
+     - "bloodmoney"
```
Advanced search uses regex, `(?i)blood money` will search for all variations of `blood money` without being case-sensitive

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+:::
+
+
+#### 6. **Focus on the Right File**: 
```
6?? Where did 6 come from

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+
+   :::info Reference Tip
+   If you come across `Line` or `dlge` files, use the **Reference List** tab to trace them back to the correct **LOCR File**.
+   :::
```
This doesn't seem compatible with the line prior which tells you to skip them, but otherwise this is much more concise

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+- `"russian"`
+- `"chineseSimplified"`
+- `"chineseTraditional"`
+- `"japanese"`
```
The suggestions in VS Code/GlacierKit also have this list

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
```diff
@@ -0,0 +1,666 @@
+
+<details>
+<summary>**Glossary**</summary>
+### Glossary 
```
You don't exactly need a glossary when literally every key word possible is in bold, italic and has a tooltip

atampy25 commented on docs/modding/suitmodding/addinglocals.mdx:
````diff
@@ -0,0 +1,666 @@
+       }
+   }
+   ```
+5. **Add Translations**: Extend the language array as needed.
````
What is the language array

atampy25 commented on docs/modding/suitmodding/customhover.css:
```diff
@@ -0,0 +1,114 @@
+  display: inline-block;
+  width: 12px;
+  height: 12px;
+  background-image: url('./exclam white.png'); /* Your image path */
```
Typo in filename

atampy25 commented on package.json:
```diff
@@ -24,12 +24,15 @@
         "@easyops-cn/docusaurus-search-local": "^0.44.5",
         "@mdx-js/react": "^3.0.0",
         "@svgr/webpack": "^6.2.1",
+        "@tippyjs/react": "^4.2.6",
```
Is tippy necessary? It feels like docusaurus would have something for this

NeetBux-Hash replied:
I looked at the other articles on the site and couldn't find anything like it

atampy25 commented on package.json:
```diff
@@ -24,12 +24,15 @@
         "react-dom": "^18.2.0",
+        "react-markdown": "^9.0.1",
         "react-responsive-carousel": "^3.2.23",
+        "react-tooltip": "^5.28.0",
```
These two dependencies appear to be useless

NeetBux-Hash commented:
The copies are unfinished drafts where I want to try to change the flow of information. I'm trying to keep a balance to prevent information dumping too much technical jargon at certain points. The problem it can do is make the articles not flow as well but it's easier to follow along. Hard to revisit tho
